pip/commands/list.py

```diff
@@ -91,6 +91,13 @@ def __init__(self, *args, **kw):
             help="Align package names and versions into vertical columns.",
         )
 
+        cmd_opts.add_option(
+            '--leaves',
+            action='store_true',
+            dest='leaves',
+            help="List packages that are not dependencies of installed packages.",
```

Comment:
Perhaps I misunderstood the requirements and I definitely am not familiar with all underlying technology behind the `pip`, but let's consider an example. 
I have six packages: 1, 2, 3, 4, 5, 6 with following dependencies:
1 -> []
2 -> []
3 -> []
4 -> [1, 2, 3]
5 -> [2, 3]
6 -> []
Packages with no dependencies are [1, 2, 3, 6] and packages which aren't depended on by anything else are [1, 2, 3, 6].